### PR TITLE
refact(xcontext): Publish package "fields"

### PR DESCRIPTION
**Pull request by xaionaro**
Moving `fields` outside of `internal`. It was required to make a fork of `bundles` (in a private repository) for `xcontext`.

### CODE REVIEW

File: pkg/xcontext/bundles/logrusctx/compact_text_formatter.go [hunk 1]
```diff
@@ -7,14 +7,14 @@ import (
...
 	"github.com/sirupsen/logrus"
 )
 
-var logLevelSymbol [logger.EndOfLevel]byte
+var logLevelSymbol []byte
 
 func init() {
-	for level := logger.Level(0); level < logger.EndOfLevel; level++ {
+	logLevelSymbol = make([]byte, len(logrus.AllLevels)+1)
+	for _, level := range logrus.AllLevels {
```
Comment (xaionaro) [inline pkg/xcontext/bundles/logrusctx/compact_text_formatter.go:17]: I previously mixed up different types of log levels. Fixing it.

**Review by tfg13 — APPROVED** [review]

**Review by rojer9-fb — APPROVED** [review]

**Comment by rojer9-fb** [comment]
test failure is a flake, i will take a look separately.

**Comment by xaionaro** [comment]
@rojer9-fb : I guess it might be caused by the massive change in `xcontext`. But I do not understand how to reproduce the problem:
```
xaionaro@void:~/go/src/github.com/facebookincubator/contest$ go test ./plugins/teststeps/... -count=5
ok      github.com/facebookincubator/contest/plugins/teststeps  21.687s
?       github.com/facebookincubator/contest/plugins/teststeps/cmd      [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/echo     [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/example  [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/randecho [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/sshcmd   [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/terminalexpect   [no test files]
xaionaro@void:~/go/src/github.com/facebookincubator/contest$ go test ./plugins/teststeps/... -count=5
ok      github.com/facebookincubator/contest/plugins/teststeps  21.687s
?       github.com/facebookincubator/contest/plugins/teststeps/cmd      [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/echo     [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/example  [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/randecho [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/sshcmd   [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/terminalexpect   [no test files]
xaionaro@void:~/go/src/github.com/facebookincubator/contest$ GOMAXPROCS=1 go test ./plugins/teststeps/... -count=5
ok      github.com/facebookincubator/contest/plugins/teststeps  21.684s
?       github.com/facebookincubator/contest/plugins/teststeps/cmd      [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/echo     [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/example  [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/randecho [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/sshcmd   [no test files]
?       github.com/facebookincubator/contest/plugins/teststeps/terminalexpect   [no test files]
```

Do you have an idea how to reproduce the failure?

**Comment by rojer9-fb** [comment]
add -race (it changes timing) and increase -count.
other than that - just do more runs on CI, it uses a virtual machine so its timings are inherently different from running locally.
patch up [tests.sh](https://github.com/facebookincubator/contest/blob/master/docker/contest/tests.sh) your repo so it only runs the test in question, with large -count, so the runs are shorter

**Pull request merged**